youtube_dl/extractor/pandoratv.py

```diff
@@ -0,0 +1,47 @@
+# encoding: utf-8
+from __future__ import unicode_literals
+
+
+from .common import InfoExtractor
+
+from ..compat import (
+    compat_urlparse,
+)
+from ..utils import (
+    ExtractorError,
+)
+
+
+class PandoraTVIE(InfoExtractor):
+    _VALID_URL = r'http://(?:.+?\.)?channel.pandora.tv/channel/video.ptv\?.*?prgid=(?P<prgid>.+?)&.*'
+    _TESTS = [{
+        'url': 'http://jp.channel.pandora.tv/channel/video.ptv?c1=&prgid=53294230&ch_userid=mikakim&ref=main&lot=cate_01_2',
+        'info_dict': {
+            'id': '53294230',
+            'ext': 'flv',
+            'title': "\u982d\u3092\u64ab\u3067\u3066\u304f\u308c\u308b\uff1f",
+        }
+    }]
+
+    def _real_extract(self, url):
+        qs = compat_urlparse.parse_qs(compat_urlparse.urlparse(url).query)
+        video_id = qs.get('prgid', [None])[0]
+        user_id = qs.get('ch_userid', [None])[0]
+        if any(not f for f in (video_id, user_id,)):
+            raise ExtractorError('Invalid URL', expected=True)
+
+        webpage = self._download_webpage(url, video_id)
+        video_set = self._html_search_regex(
+                r'var videoSet = ({.*?});',
+                webpage, 'video set')
+        video_set = self._parse_json(video_set, None)
+        info_url = "http://channel.pandora.tv/channel/video.ptv?m=getSmartData&ch_userid={userid}&prgid={prgid}".format(**video_set)
```

Comment:
You've already extracted `prgid` and `ch_userid` from URL, you should not additionally rely on these fields in `video_set`.